src/bus.c

```diff
@@ -49,50 +40,36 @@ void bus_i2c_write(uint8_t device, uint8_t reg, uint8_t value) {
     i2c_write_blocking(i2c1, device, data, 2, false);
 }
 
-void bus_i2c_io_init() {
-    printf("INIT: I2C IO\n");
-    bus_i2c_write(I2C_IO_0, I2C_IO_REG_POLARITY,   0b11111111);
-    bus_i2c_write(I2C_IO_0, I2C_IO_REG_POLARITY+1, 0b11111111);
-    bus_i2c_write(I2C_IO_1, I2C_IO_REG_POLARITY,   0b11111111);
-    bus_i2c_write(I2C_IO_1, I2C_IO_REG_POLARITY+1, 0b11111111);
-    bus_i2c_write(I2C_IO_0, I2C_IO_REG_PULLUP,     0b11111111);
-    bus_i2c_write(I2C_IO_0, I2C_IO_REG_PULLUP+1,   0b11111111);
-    bus_i2c_write(I2C_IO_1, I2C_IO_REG_PULLUP,     0b11111111);
-    bus_i2c_write(I2C_IO_1, I2C_IO_REG_PULLUP+1,   0b11111111);
-    printf("  I2C_IO_0 ");
-    printf("ack=%i ", bus_i2c_acknowledge(I2C_IO_0));
-    printf("polarity=%i ", bin(bus_i2c_read_one(I2C_IO_0, I2C_IO_REG_POLARITY)));
-    printf("pullup=%i\n", bin(bus_i2c_read_one(I2C_IO_0, I2C_IO_REG_PULLUP)));
-    printf("  I2C_IO_1 ");
-    printf("ack=%i ", bus_i2c_acknowledge(I2C_IO_1));
-    printf("polarity=%i ", bin(bus_i2c_read_one(I2C_IO_1, I2C_IO_REG_POLARITY)));
-    printf("pullup=%i\n", bin(bus_i2c_read_one(I2C_IO_1, I2C_IO_REG_PULLUP)));
+uint8_t bus_i2c_io_tristate(uint8_t index) {
+    bus_i2c_write(I2C_IO_0, I2C_IO_REG_PULL_DIR+1, 0b00000000);
+    bool down = bus_i2c_io_read(I2C_IO_0, index);
+    bus_i2c_write(I2C_IO_0, I2C_IO_REG_PULL_DIR+1, 0b11111111);
+    bool up = bus_i2c_io_read(I2C_IO_0, index);
+    if ( up !=  down) return 0;  // Float.
+    if (!up && !down) return 1;  // GND.
+    if ( up &&  down) return 2;  // VCC.
```

Comment:
It would be better to use an enum as return type, rather than literals.
Enums provide better readability, require no extra documentation (with good naming), provide certain amount of type safety, and static analyzers check for missed execution branches when used for flow control (e.g. switch statement).